R/Steph_Function_Attempts.R

```diff
@@ -0,0 +1,101 @@
+mnTest <- dataone::getMNode(cnTest,'urn:node:mnTestARCTIC')
+
+#dummy package
+pkg <- arcticdatautils::create_dummy_package(mnTest,
+                                             size = 2)
+#dummy data table:
+attributes1 <- data.frame(
+    attributeName = c('col1', 'col2'),
+    attributeDefinition = c('Numbers', 'Letters in the alphabet'),
+    measurementScale = c('ratio', 'nominal'),
+    domain = c('numericDomain', 'textDomain'),
+    formatString = c(NA,NA),
+    definition = c(NA,'ABCDEFG...'),
+    unit = c('dimensionless', NA),
+    numberType = c('integer', NA),
+    missingValueCode = c(NA, NA),
+    missingValueCodeExplanation = c(NA, NA),
+    stringsAsFactors = FALSE)
+
+attributeList1 <- set_attributes(attributes1)
+phys <- pid_to_eml_physical(mnTest, pkg$data[1])
+
+dummy_data_table <- new('dataTable',
+                        entityName = 'Dummy Data Table',
+                        entityDescription = 'Dummy Description',
+                        physical = phys,
+                        attributeList = attributeList1)
+
+eml <- read_eml(rawToChar(getObject(mnTest, pkg$metadata)))
+eml@dataset@dataTable <- c(dummy_data_table)
+
+pkg #getting PIDs
+# $metadata
+# [1] "urn:uuid:83965425-2125-4663-bac7-d471fb03fb43"
+# $resource_map
+# [1] "urn:uuid:3a26404a-a8d0-4a6b-b081-3d6d96b40923"
+# $data
+# [1] "urn:uuid:9f1a87e5-0c1a-4498-b787-afe40d417be2"
+
+eml@dataset@dataTable@.Data
+#Need some way to access the <attributeList> inside this slot: maybe write in the xml text?
+
+#arguments would be listed like so, where x is the attribute of interest
+#AttName<-edit_attribute(x, attributeName="",attributeLabel="",attributeDefinition="",measurementScale="",
+                            #domain="",formatString="",definition="",unit="",numberType="",
+                            #missingValueCode="",missingValueCodeExplanation="")
+
+##Don't use this
+# edit_attribute<-function(attributeName="",attributeDefinition="",measurementScale="",
+#                          domain="",formatString="",definition="",unit="",numberType="",
+#                          missingValueCode="",missingValueCodeExplanation=""){
+#     paste0('<attribute>
+#              <attributeName>',attributeName,'</attributeName>',
+#              '<attributeDefinition>',attributeDefinition,'</attributeDefinition>',
+#              '<measurementScale><',measurementScale,'>',
+#                 '<unit><standardUnit>',unit,'</standardUnit></unit>',
+#                 '<',domain,'>','<numberType>',numberType,'</numberType></',domain,'>',
+#              '</measurementScale></attribute>')
+# }
+
+#Code below includes only the required slots and does not contain any if statements for failing the function if e.g. the wrong domain goes into a measurement scale.
+#x is the data table number in the data table list, y is the attribute number in the attribute list for that data table
+edit_attribute<-function(x,y,attributeName,attributeDefinition,measurementScale,
```

Comment:
It would be nice to not require the user to submit all of these elements of an attribute, just in case the user just wants to change only one thing, eg the attributeName.